### PR TITLE
EYQB-500: Updated logic to add in new error messages for the When was the qualification started page

### PR DESCRIPTION
**Pull request by sam-c-dfe**
# Description

Updated logic to add in new error messages for the When was the qualification started page

## Ticket number (if applicable) - EYQB-500

# How Has This Been Tested?

Running locally, unit tests and e2e tests

# Screenshots

image

image

image

# Checklist:

- [x] My code follows the standards used within this project
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests (Unit, E2E) that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by DanielClarkeEducation — DISMISSED** [review]

**Review by DanielClarkeEducation — APPROVED** [review]

File: src/Dfe.EarlyYearsQualification.Web/Helpers/PlaceholderUpdater.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+namespace Dfe.EarlyYearsQualification.Web.Helpers;
+
+public class PlaceholderUpdater : IPlaceholderUpdater
+{
+    private readonly Dictionary<string, string> _placeholders = new()
+                                                                {
+                                                                    { "$[actual-year]$", DateTimeOffset.Now.Year.ToString() }
```
Comment (RobertGHippo) [inline src/Dfe.EarlyYearsQualification.Web/Helpers/PlaceholderUpdater.cs:7]: Suggestion (minor). Single responsibility: I'd inject an `IDateTimeAdapter` instance into this service, and get the year from that? That might make unit testing easier, particularly if we ever need to extend the functionality to other placeholders. _This_ service's responsibility is only to replace placeholders. Actually, we could then replace this dictionary with a static array of placeholders.  🙂

File: src/Dfe.EarlyYearsQualification.Web/Helpers/PlaceholderUpdater.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                                                                };
+
+    public string Replace(string valueToCheck)
```
Comment (RobertGHippo) [inline src/Dfe.EarlyYearsQualification.Web/Helpers/PlaceholderUpdater.cs:10]: Suggestion (minor). It's not really doing a "check". How about just naming this parameter `text`?

File: tests/Dfe.EarlyYearsQualification.UnitTests/Helpers/PlaceholderUpdaterTests.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+using Dfe.EarlyYearsQualification.Web.Helpers;
+using FluentAssertions;
+
+namespace Dfe.EarlyYearsQualification.UnitTests.Helpers;
+
+[TestClass]
+public class PlaceholderUpdaterTests
```
Comment (RobertGHippo) [inline tests/Dfe.EarlyYearsQualification.UnitTests/Helpers/PlaceholderUpdaterTests.cs:7]: Suggestion (minor): might be worth a test to check that if the text contains multiple copies of the placeholder then all of them are replaced? `"The year $[actual-year] is $[actual-year]$"`. 🙂

**Review by RobertGHippo — DISMISSED** [review]

File: src/Dfe.EarlyYearsQualification.Web/Helpers/PlaceholderUpdater.cs (hunk 2 above)
Comment (sam-c-dfe) [reply src/Dfe.EarlyYearsQualification.Web/Helpers/PlaceholderUpdater.cs]: Yep, wasn't sure on the name. Text is a better suggestion :-)

File: tests/Dfe.EarlyYearsQualification.UnitTests/Helpers/PlaceholderUpdaterTests.cs (hunk 3 above)
Comment (sam-c-dfe) [reply tests/Dfe.EarlyYearsQualification.UnitTests/Helpers/PlaceholderUpdaterTests.cs:9]: Added that now :-)

File: src/Dfe.EarlyYearsQualification.Web/Helpers/PlaceholderUpdater.cs (hunk 1 above)
Comment (sam-c-dfe) [reply src/Dfe.EarlyYearsQualification.Web/Helpers/PlaceholderUpdater.cs]: Injected the adapter now :-)

**Review by RobertGHippo — APPROVED** [review]

**Pull request merged**